Add edit appointment command

caitlyntang opened the pull request:
Fixes: #149 

### Describe your changes
Adds new edit appointment command. Command: `edita INDEX [pid/] [sd/]` etc.

tankh99 reviewed (changes requested):
Just a couple of bugs I found:
1. You're able to bypass the start date must be before end date validation via edit appointment
<img width="1316" alt="Screenshot 2024-04-03 at 11 28 16 AM" src="https://github.com/AY2324S2-CS2103-F08-2/tp/assets/56617918/5c2a86d2-0307-4759-a12c-4b1031041b39">
2. There is a strange discrepancy when editing description vs score. When I leave `ad/`  blank, I manage to set the description as null. However, when I leave `s/` blank, The score isn't changed. 
<img width="1312" alt="Screenshot 2024-04-03 at 11 37 54 AM" src="https://github.com/AY2324S2-CS2103-F08-2/tp/assets/56617918/ad3bd663-8ae3-40d2-9976-8c062300e9d2">

caitlyntang commented:
@tankh99 Thanks for spotting the bugs.

1. I have added more checks for time validation
2. This issue was caused by appointment description having an empty string as the default value. I have changed it to null so the result should be that the value is not changed when left blank. This applies to prefixes `s/`, `ad/` and `att/`.

Please test again. Thanks!

tankh99 reviewed (changes requested):
I think the way you check for start and end date times currently is not really correct? I tried editing the start date time to be 1 min after its original and it fails. In fact, it seems that it fails even if I set the start date time to 1hr before, even though there is no appointment that is at that time. Perhaps consider writing a new method rather than use the RelationshipUtil method, because it was used to check only one specific date last time

<img width="1310" alt="Screenshot 2024-04-04 at 6 37 09 AM" src="https://github.com/AY2324S2-CS2103-F08-2/tp/assets/56617918/90f85bd0-235e-4687-b760-e2fef281d85c">

Regarding the nullable description, I have tested and it works consistently now

caitlyntang commented:
Noted, I have tried to improve the time checking algo. Please test again

tankh99 reviewed (approved):
Thanks Caitlyn, the previous bugs mentioned no longer appear. Now there is a minor inconsistency here. The error message is not entirely accurate since the fields are actually quite different. I suspect the message should instead be more like "There is already an appointment at that time"

However, most of the issues seem to be resolved now, so I'll approve

<img width="1340" alt="Screenshot 2024-04-04 at 10 29 21 AM" src="https://github.com/AY2324S2-CS2103-F08-2/tp/assets/56617918/17652371-f648-4dec-929d-a287c69e4c90">
